### PR TITLE
Fix `\fp_clear_function:n` acting globally, add tests, and a bit more

### PR DESCRIPTION
**Pull request by muzimuzhi**
I'd like to recheck content of new `m3fp-symbolic002.tlg` file before merging.

### CODE REVIEW

File: l3kernel/l3fp-functions.dtx [hunk 1]
```diff
@@ -180,7 +180,7 @@
           \exp_args:No \clist_map_inline:nn { \tl_to_str:n {#3} }
             {
               \int_incr:N \l_@@_function_arg_int
-              \exp_args:Ne \@@_clear_variable:n
+              \exp_args:Ne \@@_clear_variable_aux:n
                 { _ \tex_romannumeral:D \l_@@_function_arg_int }
```
Comment (muzimuzhi) [inline l3kernel/l3fp-functions.dtx:184]: `_ \tex_romannumeral:D \l_@@_function_arg_int` is not a str.

File: l3kernel/l3fp-functions.dtx [hunk 2]
```diff
@@ -98,7 +98,7 @@
   }
 \cs_new:Npn \@@_function_set_parsing_aux:NNn #1#2#3
   {
-    \cs_set:Npe \@@_tmp:w
+    \cs_set_nopar:Npe \@@_tmp:w
```
Comment (josephwright) [inline l3kernel/l3fp-functions.dtx:101]: Why `nopar` here? Other than for vars and for `ltcmd` interfaces, we stick to `\cs_set:Npn`, etc.

File: l3kernel/l3fp-functions.dtx [hunk 3]
```diff
@@ -180,8 +180,10 @@
           \exp_args:No \clist_map_inline:nn { \tl_to_str:n {#3} }
             {
               \int_incr:N \l_@@_function_arg_int
-              \exp_args:Ne \@@_clear_variable:n
-                { _ \tex_romannumeral:D \l_@@_function_arg_int }
+              \exp_args:Ne \@@_clear_variable_aux:n
+                {
+                  \tl_to_str:e { _ \tex_romannumeral:D \l_@@_function_arg_int }
```
Comment (josephwright) [inline l3kernel/l3fp-functions.dtx:185]: Do we need `\tl_to_str:n` here? Isn't the internal path directly to a csname? The `_` must have catcode 11 whilst the roman numeral generation gives catcode 12.

**Review by josephwright — CHANGES_REQUESTED** [review]

File: l3kernel/l3fp-functions.dtx (hunk 2 above)
Comment (muzimuzhi) [reply l3kernel/l3fp-functions.dtx:101]: You're right, I should do in the reverse direction --- dropping `nopar` in `l3fp-symbolic.dtx`.

File: l3kernel/l3fp-functions.dtx (hunk 3 above)
Comment (muzimuzhi) [reply l3kernel/l3fp-functions.dtx:185]: I'll change to `\c_underscore_str \tex_romannumeral:D \l_@@_function_arg_int`, to stick to the assumption that all first-args of internal forms `\@@_(new|set|clear)_(variable|function):...` are str.

**Review by josephwright — APPROVED** [review]

**Comment by muzimuzhi** [comment]
Commits rebased. Ready to merge.

**Pull request merged**